### PR TITLE
add rule for store connections deps

### PR DESCRIPTION
**Pull request by jixwanwang**
that was fun! hopefully reasonable code.

i could use help with descriptions and such, i'm bad at words. I will also write some more test cases on monday.

### CODE REVIEW

**Comment by meishuu** [comment]
wait shit.

we should archive this repo b/c we brought it into the monorepo lol

https://github.com/discord/discord/tree/master/discord_common/js/packages/eslint-plugin-discord

File: lib/rules/store-connect-deps.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            });
+          }
+        });
+
+        // Get all stores that are used in the function body
+        let actualDeps = [];
+
+        traverse(context, args[1], (path) => {
+          let node = path.node;
+          if (node.type !== 'MemberExpression' || node.object.type !== 'Identifier') {
```
Comment (jamiebuilds) [inline lib/rules/store-connect-deps.js:73]: Would it be better to just look for Identifiers with the name ending in `"Store"`? `traverse` can walk through the MemberExpression.

Comment (jamiebuilds) [reply lib/rules/store-connect-deps.js:73]: Just thinking about cases like: `selectDataFrom(MyStore)`

The safest way to make sure that you're referencing a store is to look up the identifier's binding in scope.

Comment (jixwanwang) [reply lib/rules/store-connect-deps.js:73]: that's a good point, that will probably work better. the counterpoint is if someone assigns a variable like dataFromStore, but it'd be pretty easy to recognize that issue

**Comment by jixwanwang** [comment]
moving to the monorepo https://github.com/discord/discord/pull/26329

**Pull request closed without merging**